openforcefield/tests/utils.py

```diff
@@ -897,42 +1047,67 @@ def compare_system_parameters(system1, system2):
     assert set(force_names2.values()) == {1}, err_msg
 
+    # Remove the center-of-mass motion remover force from comparison.
+    for force_names in [force_names1, force_names2]:
+        if 'CMMotionRemover' in force_names:
+            del force_names['CMMotionRemover']
+
     # Check that the two systems have the same forces.
-    err_msg = 'The two Systems have different forces: system1 {}, system2 {}'
+    err_msg = 'The two Systems have different forces to compare: system1 {}, system2 {}'
     assert set(force_names1) == set(force_names2), err_msg.format(force_names1, force_names2)
 
     # Find all the pair of forces to compare.
     force_pairs = {force_name: [] for force_name in force_names1}
     for system in [system1, system2]:
         for force in system.getForces():
+            # Skip CMMotionRemover.
+            if force.__class__.__name__ == 'CMMotionRemover':
+                continue
             force_pairs[force.__class__.__name__].append(force)
 
     # Compare all pairs of forces
     for force_name, (force1, force2) in force_pairs.items():
-        parameters_force1 = _get_force_parameters(force1, system1)
-        parameters_force2 = _get_force_parameters(force2, system2)
+        # Select the parameters to ignore.
+        try:
+            ignored_parameters = ignored_parameters_by_force[force_name]
+        except KeyError:
+            ignored_parameters = ()
+
+        # Get a dictionary representation of the parameters.
+        parameters_force1 = _get_force_parameters(force1, system1, ignored_parameters)
+        parameters_force2 = _get_force_parameters(force2, system2, ignored_parameters)
+
+        # Compare the forces and raise an error if a difference is detected.
         for parameter_type, parameters1 in parameters_force1.items():
             parameters2 = parameters_force2[parameter_type]
             _compare_parameters(parameters1, parameters2,
-                                interaction_type=parameter_type)
+                                interaction_type=parameter_type,
+                                force_name=force_name,
+                                systems_labels=systems_labels)
 
 
 #=============================================================================================
 # Utility functions to compare SMIRNOFF and AMBER force fields.
 #=============================================================================================
 
-def compare_amber_smirnoff(prmtop_filepath, inpcrd_filepath, forcefield, molecule):
+def compare_amber_smirnoff(prmtop_filepath, inpcrd_filepath, forcefield, molecule,
+                           check_parameters=True, check_energies=True, **kwargs):
     """
     Compare energies and parameters for OpenMM Systems/topologies created
     from an AMBER prmtop and crd versus from a SMIRNOFF forcefield file which
-    should parameterize the same system with same parameters.
+    should parametrize the same system with same parameters.
```

Comment:
"parameterize", I think

Reply:
https://en.oxforddictionaries.com/definition/parameterize
https://english.stackexchange.com/questions/79847/parametrise-or-parameterise-a-curve

Looks like it's still an open question. Also it could end in "ise" in British english. Looks like, in addition to being in a scientific niche, we're also in a linguistic one. 

I have no strong opinion on which spelling we use, but for consistency we should agree on one (otherwise we'll have N+1 potential coding errors). @andrrizzi, if you have no preference, I'd suggest "parameterize". @SimonBoothroyd, will this be inconvenient for you?

Reply:
I think it's a good idea to stick to one convention. If @SimonBoothroyd is on board too, I'll go ahead and fix it before merging.